contracts/contracts/vault/VaultCore.sol

```diff
@@ -267,12 +267,22 @@ contract VaultCore is VaultStorage {
             IStrategy strategy = IStrategy(allStrategies[i]);
             address rewardTokenAddress = strategy.rewardTokenAddress();
             if (rewardTokenAddress != address(0)) {
-                IERC20 rewardToken = IERC20(rewardTokenAddress);
-                uint256 rewardTokenAmount = rewardToken.balanceOf(
-                    address(this)
-                );
-                if (rewardTokenAmount > strategy.rewardLiquidationThreshold()) {
+                uint256 liquidationThreshold = strategy.rewardLiquidationThreshold();
+                if (liquidationThreshold == 0) {
+                    // No threshold set, always harvest from strategy
                     IVault(address(this)).harvest(allStrategies[i]);
+                } else {
+                    // Check balance against liquidation threshold
+                    // Note some strategies don't hold the reward token balance
+                    // on their contract so the liquidation threshold should be
+                    // set to 0
+                    IERC20 rewardToken = IERC20(rewardTokenAddress);
+                    uint256 rewardTokenAmount = rewardToken.balanceOf(
```

Comment:
Not specific to this PR, but do I understand correctly that the threshold must be met at the strategy level regardless of aggregate token balance? In other words, if we had a CompoundDAI strategy and a CompoundUSDT strategy with each holding 0.75 COMP, neither would get liquidated. And then if one of them is 1.25, it will get liquidated while the other won't if its balance is still below the threshold?

Reply:
Yep that is correct!